src/Worker.Extensions.DurableTask/DurableTaskClientExtensions.cs

```diff
@@ -143,12 +172,31 @@ static string BuildUrl(string url, params string?[] queryValues)
         //       request headers into consideration and generate the base URL accordingly.
         //       More info: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded.
         //       One potential workaround is to set ASPNETCORE_FORWARDEDHEADERS_ENABLED to true.
-        string baseUrl = request.Url.GetLeftPart(UriPartial.Authority);
+
+        // If HttpRequestData is provided, use its URL; otherwise, get the baseUrl from the DurableTaskClient.
+        // The base URL could be null if:
+        // 1. The DurableTaskClient isn't a FunctionsDurableTaskClient (which would have the baseUrl from bindings)
+        // 2. There's no valid HttpRequestData provided
+        string? baseUrl = ((request != null) ? request.Url.GetLeftPart(UriPartial.Authority) : GetBaseUrl(client))
+            ?? throw new InvalidOperationException("Base URL is null. Either use Functions bindings or provide an HTTP request to create the HttpPayload.");
+        bool isFromRequest = request != null;
+
         string formattedInstanceId = Uri.EscapeDataString(instanceId);
-        string instanceUrl = $"{baseUrl}/runtime/webhooks/durabletask/instances/{formattedInstanceId}";
+
+        // The baseUrl differs depending on the source. Eg:
+        // - From request: http://localhost:7071/
```

Comment:
I'm not sure I understand what it means to be "from request". Can you please clarify?

Reply:
There are two ways to obtain the BaseUrl: (1) from HttpRequestData, which refers to the request in this context. And this usually happen when cx already have a Http bindings, or (2) from the Functions host, if HttpRequestData is not provided, as this information is passed through the DurableClient bindings.

Let me try to update this comment to be more clear. 

Eg, calling CreateHttpManagementPayload with HttpRequestData like this:

![image](https://github.com/user-attachments/assets/79120f57-0cc4-46fd-96fd-d90059e02846)